.github/workflows/release.yml

```diff
@@ -47,60 +54,69 @@ jobs:
             ;;
           esac
           tag="v${release_version}"
           {
             echo "branch_name=${branch_name}"
             echo "release_version=${release_version}"
             echo "release_development_version=${release_development_version}"
+            echo "master_development_version=${master_development_version}"
             echo "tag=${tag}"
           } >> "$GITHUB_OUTPUT"
+
       - if: ${{ inputs.bump == 'major' || inputs.bump == 'minor' }}
-        name: 'Release ${{ inputs.bump }} version ${{ steps.compute_versions.outputs.release_version }} on branch ${{ steps.compute_versions.outputs.branch_name }}'
+        name: 'Create ${{ inputs.bump }} release branch ${{ steps.compute_versions.outputs.branch_name }}'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           ./mvnw release:clean release:branch \
               -DbranchName=${{ steps.compute_versions.outputs.branch_name }} \
               -DreleaseVersion=${{ steps.compute_versions.outputs.release_version }} \
-              -DdevelopmentVersion=${{ steps.compute_versions.outputs.release_development_version }} \
-              -Darguments='-DskipTests=true'
-          ./mvnw release:prepare release:perform \
-              -DupdateBranchVersions=false \
-              -DupdateWorkingCopyVersions=false \
-              -Dtag=${{ steps.compute_versions.outputs.tag }} \
-              -DreleaseVersion=${{ steps.compute_versions.outputs.release_version }} \
-              -DdevelopmentVersion=${{ steps.compute_versions.outputs.release_development_version }} \
-              -Darguments='-DskipTests=true'
-      - if: ${{ inputs.bump == 'hotfix' }}
-        name: 'Release ${{ inputs.bump }} version ${{ steps.compute_versions.outputs.release_version }} on branch ${{ steps.compute_versions.outputs.branch_name }}'
+              -DdevelopmentVersion=${{ steps.compute_versions.outputs.master_development_version }}
+
+      - if: ${{ inputs.bump == 'major' || inputs.bump == 'minor' }}
+        name: 'Checkout ${{ inputs.bump }} release branch ${{ steps.compute_versions.outputs.branch_name }}'
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ steps.compute_versions.outputs.branch_name }}
+
+      - name: 'Prepare release ${{ inputs.bump }} version ${{ steps.compute_versions.outputs.release_version }} on branch ${{ steps.compute_versions.outputs.branch_name }}'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          ./mvnw release:clean release:prepare release:perform \
-              -Dtag=${{ steps.compute_versions.outputs.tag }} \
+          ./mvnw release:clean release:prepare \
               -DreleaseVersion=${{ steps.compute_versions.outputs.release_version }} \
               -DdevelopmentVersion=${{ steps.compute_versions.outputs.release_development_version }} \
-              -Darguments='-DskipTests=true'
+              -Dtag=${{ steps.compute_versions.outputs.tag }}
```

Comment:
The actual release preparation process should be shared between major, minor, and hotfix releases, because it always happens onto the release branch.

The version computation is what actually changes (besides the need to start a release branch for major and minor releases); I thought about tracking it into Maven profiles, but it was not easily feasible (if at all).